### PR TITLE
DEVPROD-2731 Introduce task displayStatus field

### PR DESCRIPTION
**Pull request by khelif96**
DEVPROD-2731

### Description
This is the first pr in a series of prs to refactor the `status` type so that is reflects the task's original status and add a new field to reflect the `displayStatus`.
This will be used to show the original status alongside the display status in the UI.

### Testing
GraphQL tests
and UI PR

This PR allows us to display both the known issue status and display status with the linked UI pr
image

### CODE REVIEW

File: graphql/schema/types/task.graphql [hunk 1]
```diff
@@ -109,6 +113,9 @@ type Task {
   scheduledTime: Time
   spawnHostLink: String
   startTime: Time
+  """
+  This is a tasks display status and is what is commonly used on the UI.
```
Comment (sophstad) [inline graphql/schema/types/task.graphql:117]:
```suggestion
  This is a task's display status and is what is commonly used on the UI.
```

File: graphql/schema/types/task.graphql [hunk 2]
```diff
@@ -96,6 +96,10 @@ type Task {
   logs: TaskLogLinks!
   minQueuePosition: Int!
   order: Int!
+  """
+  The originalStatus field represents the original status of the task without a display status applied.
+  """
+  originalStatus: String!
```
Comment (sophstad) [inline graphql/schema/types/task.graphql:102]: If you're willing to do this in a backwards-compatible way I feel like it'd be preferable to have this align with the rest of the codebase, i.e. using `status` and `displayStatus`. Not a blocker though as I imagine that could be a wide-reaching change on the front end.

File: model/task/task.go [hunk 3]
```diff
@@ -2334,6 +2334,9 @@ func (t *Task) findDisplayStatus() string {
 	if t.Status == evergreen.TaskSucceeded {
 		return evergreen.TaskSucceeded
 	}
+	if t.HasAnnotations {
+		return evergreen.TaskKnownIssue
+	}
```
Comment (sophstad) [inline model/task/task.go:2339]: I believe this should be equivalent to the [agg pipeline](https://github.com/evergreen-ci/evergreen/blob/30b1913586d8e0bbf2862a41231a6aae14f0f039/model/task/db.go#L215-L218) which has Known Issue as the status of highest priority

**Review by sophstad — CHANGES_REQUESTED** [review]

**Review by Kimchelly — APPROVED** [review]
LGTM mod [the same comment about moving the known issue check](https://github.com/evergreen-ci/evergreen/pull/8529/files#r1871984117).

File: graphql/schema/types/task.graphql (hunk 2 above)
Comment (khelif96) [reply graphql/schema/types/task.graphql]: I'm starting this now with this pr. A follow up UI pr will swap fields to use this new field. And in a future pr `status` will go back to its originall value.

File: graphql/schema/types/task.graphql (hunk 1 above)
Comment (sophstad) [reply graphql/schema/types/task.graphql:114]: Bump

File: graphql/task_resolver.go [hunk 4]
```diff
@@ -644,3 +644,13 @@ func (r *taskResolver) VersionMetadata(ctx context.Context, obj *restModel.APITa
...
+
+// !!! WARNING !!!
+// The code below was going to be deleted when updating resolvers. It has been copied here so you have
+// one last chance to move it out of harms way if you want. There are two reasons this happens:
+//   - When renaming or deleting a resolver the old code will be put in here. You can safely delete
+//     it when you're done.
+//   - You have helper methods in this file. Move them out to keep these resolver files clean.
+func (r *taskResolver) OriginalStatus(ctx context.Context, obj *restModel.APITask) (string, error) {
+	return *obj.Status, nil
+}
```
Comment (sophstad) [inline graphql/task_resolver.go:656]: Delete

File: graphql/tests/task/originalStatus/queries/original_status.graphql [hunk 5]
```diff
@@ -0,0 +1,9 @@
+query {
+  task(
+    taskId: "evergreen_ubuntu1604_test_db_patch_5e823e1f28baeaa22ae00823d83e03082cd148ab_5e4ff3abe3c3317e352062e4_20_02_21_15_13_48"
+    execution: 0
+  ) {
+    status
+    originalStatus
```
Comment (sophstad) [inline graphql/tests/task/originalStatus/queries/original_status.graphql:7]: This test should be updated

File: model/task/task.go [hunk 6]
```diff
@@ -2328,11 +2328,15 @@ func (t *Task) GetDisplayStatus() string {
...
 		return t.DisplayStatus
 	}
-	t.DisplayStatus = t.findDisplayStatus()
+	t.DisplayStatus = t.calculateDisplayStatus()
 	return t.DisplayStatus
 }
 
-func (t *Task) findDisplayStatus() string {
+// calculateDisplayStatus calculates the display status for a task based on its current state.
+func (t *Task) calculateDisplayStatus() string {
```
Comment (sophstad) [inline model/task/task.go:2336]: [nit] Could we not rename this for now? I have a lot of branches in flight for DEVPROD-12742 that would be complicated 😅 maybe a followup ticket?

**Review by sophstad — CHANGES_REQUESTED** [review]

**Review by sophstad — APPROVED** [review]
Nice! I would just suggest renaming the PR to "Introduce task display status field" or similar so that the commit message is accurate 👍

**Pull request merged**